flake.nix

```
@@ -401,25 +402,25 @@
         };
       };
 
-      # Build the Nickel VSCode extension. The extension seems to be required
-      # for the LSP to work.
-      vscodeExtension =
-        let node-package = (pkgs.callPackage ./lsp/client-extension { }).package;
-        in
-        (node-package.override rec {
-          pname = "nls-client";
-          outputs = [ "vsix" "out" ];
-          nativeBuildInputs = with pkgs; [
-            # `vsce` depends on `keytar`, which depends on `pkg-config` and `libsecret`
-            pkg-config
-            libsecret
-          ];
-          postInstall = ''
-            npm run compile
-            mkdir -p $vsix
-            echo y | npx vsce package -o $vsix/${pname}.vsix
-          '';
-        }).vsix;
+      # Build the Nickel VSCode extension
+      vscodeExtension = pkgs.mkYarnPackage {
+        pname = "vscode-nickel";
+        src = pkgs.lib.cleanSource ./lsp/client-extension;
+
+        buildPhase = ''
+          export HOME="$TMPDIR"
```

Sounds reasonable, but it might be useful to persist this information

```suggestion
          # yarn tries to create a .yarn file in $HOME. There's probably a
          # better way to fix this but setting HOME to TMPDIR works for now.
          export HOME="$TMPDIR"
```